metrics/metrics.go

```diff
@@ -53,78 +53,108 @@ func Flush() error {
 }
 
 // Gauge measures the value of a metric at a particular time.
-func Gauge(name string, value float64, options ...MetricOptions) {
-	metricOpts := parseMetricOptions(options...)
-	err := statsdClient.Gauge(name, value, metricOpts.tags, metricOpts.sampleRate)
+func Gauge(name string, value float64, options ...Option) {
+	localOpts := (*globalOpts) // Make copy of global opts to avoid mutating the global state
+	err := localOpts.applyOptions(options)
 	if err != nil {
-		opts.errorHandler(fmt.Errorf("failed to send Gauge: %w", err))
+		localOpts.errorHandler(fmt.Errorf("failed to apply metric options: %w", err))
+		return
+	}
+	err = statsdClient.Gauge(name, value, localOpts.tags, localOpts.sampleRate)
+	if err != nil {
+		globalOpts.errorHandler(fmt.Errorf("failed to send Gauge: %w", err))
```

Comment:
Should this not be opposite? Use the global error handler then the local if there is no errors settings the local options. With the current implementation the local error handler is only used to handle error caused by processing the local options and only if it was registered before `localOpts.applyOptions(options)` returns an error.
  
```suggestion
		globalOpts.errorHandler(fmt.Errorf("failed to apply metric options: %w", err))
		return
	}
	err = statsdClient.Gauge(name, value, localOpts.tags, localOpts.sampleRate)
	if err != nil {
		localOpts.errorHandler(fmt.Errorf("failed to send Gauge: %w", err))
```

Alternatively use the local error handler in both cases since all the `globalOps` are copied to the `localOpts`

```suggestion
		localOpts.errorHandler(fmt.Errorf("failed to apply metric options: %w", err))
		return
	}
	err = statsdClient.Gauge(name, value, localOpts.tags, localOpts.sampleRate)
	if err != nil {
		localOpts.errorHandler(fmt.Errorf("failed to send Gauge: %w", err))
```

Reply:
Good catch, my intention was to use the local handler both places, I'll fix it tomorrow

Reply:
Fixed